numpy/core/src/multiarray/arrayobject.c

```diff
@@ -112,7 +112,7 @@ PyArray_SetUpdateIfCopyBase(PyArrayObject *arr, PyArrayObject *base)
      * references.
      */
     ((PyArrayObject_fields *)arr)->base = (PyObject *)base;
-    PyArray_ENABLEFLAGS(arr, NPY_ARRAY_UPDATEIFCOPY);
+    PyArray_ENABLEFLAGS(arr, NPY_ARRAY_UPDATEIFCOPY_CLEAR_B4_EXIT);
```

Comment:
Sorry, I don't understand the naming scheme here. IIRC (from my phone), `PyArray_SetUpdateIfCopyBase` sets the `UPDATEIFCOPY` flag and sets the `->base` member, but it doesn't copy any data does it? I was suggesting having a version that's identical except that it sets the new flag instead of the old one and doesn't warn. What does `PyArray_CopyDataFromBase` do?